packages/chain-adapters/src/evm/EVMBaseAdapter.ts

```diff
@@ -0,0 +1,285 @@
+import { AssetId, ChainId, toAssetId } from '@shapeshiftoss/caip'
+import { ETHSignMessage } from '@shapeshiftoss/hdwallet-core'
+import { bip32ToAddressNList, ETHSignTx, ETHWallet } from '@shapeshiftoss/hdwallet-core'
+import { BIP44Params, KnownChainIds } from '@shapeshiftoss/types'
+import * as unchained from '@shapeshiftoss/unchained-client'
+import WAValidator from 'multicoin-address-validator'
+
+import { ChainAdapter as IChainAdapter } from '../api'
+import { ErrorHandler } from '../error/ErrorHandler'
+import {
+  Account,
+  BuildSendTxInput,
+  ChainTxType,
+  FeeDataEstimate,
+  GetAddressInput,
+  GetFeeDataInput,
+  SignMessageInput,
+  SignTxInput,
+  SubscribeError,
+  SubscribeTxsInput,
+  Transaction,
+  TxHistoryInput,
+  TxHistoryResponse,
+  ValidAddressResult,
+  ValidAddressResultType
+} from '../types'
+import {
+  chainIdToChainLabel,
+  getAssetNamespace,
+  getStatus,
+  getType,
+  toPath,
+  toRootDerivationPath
+} from '../utils'
+import { bnOrZero } from '../utils/bignumber'
+
+export type EVMChainIds = KnownChainIds.EthereumMainnet | KnownChainIds.AvalancheMainnet
+
+export interface ChainAdapterArgs {
+  chainId?: ChainId
+  providers: {
+    http: unchained.ethereum.V1Api
+    ws: unchained.ws.Client<unchained.ethereum.EthereumTx>
+  }
+  rpcUrl: string
+}
+
+export interface EVMBaseAdapterArgs extends ChainAdapterArgs {
+  supportedChainIds: Array<ChainId>
+  chainId: ChainId
+}
+
+export abstract class EVMBaseAdapter<T extends EVMChainIds> implements IChainAdapter<T> {
+  protected readonly chainId: ChainId
+  protected readonly supportedChainIds: Array<ChainId>
+  protected readonly providers: {
+    http: unchained.ethereum.V1Api
+    ws: unchained.ws.Client<unchained.ethereum.EthereumTx>
+  }
+
+  protected assetId: AssetId
```

Comment:
```suggestion
  protected feeAssetId: AssetId
```

i don't really mind either wya here but this seems more explicit

Reply:
I am going to leave as is since it matches the rest of the chain adapters and we can always run back through and update them all at once to match if we like `feeAssetId` better